chargebee-provider/src/main/java/org/kindlyops/providers/chargebee/ChargeBeeRegistration.java

```diff
@@ -65,8 +74,44 @@ public String getId() {
 
     @Override
     public void buildPage(FormContext context, LoginFormsProvider form) {
+        AuthenticatorConfigModel chargebeeConfig = context.getAuthenticatorConfig();
+        if (chargebeeConfig == null || chargebeeConfig.getConfig() == null
+                || chargebeeConfig.getConfig().get(API_KEY) == null
+                || chargebeeConfig.getConfig().get(PLAN_ID) == null
+                || chargebeeConfig.getConfig().get(SITE_NAME) == null
+                ) {
+            form.addError(new FormMessage(null, "Chargebee not configured."));
+            return;
+        }
+        String apiKey = chargebeeConfig.getConfig().get(API_KEY);
+        String planID = chargebeeConfig.getConfig().get(PLAN_ID);
+        String siteName = chargebeeConfig.getConfig().get(SITE_NAME);
+        Environment.configure(siteName,apiKey);
+        Result finalResult;
+        try {
+            finalResult = HostedPage.checkoutNew()
+                .subscriptionPlanId(planID)
+                .customerEmail("john@user.com")
```

Comment:
Since this is an initial registration, should these user data calls be omitted? If an already-registered user was checking out, I can see how it would be nice to pre-populate.

Reply:
I agree, (if I understand correctly, heh). I don't think we need them for registration

Reply:
Yeah we don't need them here right now.